data/hooks/diagnosis/21-web.py

```diff
@@ -43,19 +53,39 @@ def run(self):
                 raise YunohostError("diagnosis_http_could_not_diagnose", error=e)
 
             if r["status"] == "ok":
+                at_least_one_domain_ok = True
                 yield dict(meta={"domain": domain},
                            status="SUCCESS",
-                           summary=("diagnosis_http_ok", {"domain": domain}))
+                           summary="diagnosis_http_ok")
             else:
                 detail = r["code"].replace("error_http_check", "diagnosis_http") if "code" in r else "diagnosis_http_unknown_error"
                 yield dict(meta={"domain": domain},
                            status="ERROR",
-                           summary=("diagnosis_http_unreachable", {"domain": domain}),
-                           details=[(detail,())])
+                           summary="diagnosis_http_unreachable",
+                           details=[detail])
 
-        # In there or idk where else ...
-        # try to diagnose hairpinning situation by crafting a request for the
-        # global ip (from within local network) and seeing if we're getting the right page ?
+        # If at least one domain is correctly exposed to the outside,
+        # attempt to diagnose hairpinning situations. On network with
+        # hairpinning issues, the server may be correctly exposed on the
+        # outside, but from the outside, it will be as if the port forwarding
+        # was not configured... Hence, calling for example
+        # "curl --head the.global.ip" will simply timeout...
+        if at_least_one_domain_ok:
+            ipv4 = Diagnoser.get_cached_report_item("ip", {"test": "ipv4"})
+            global_ipv4 = ipv4.get("data", {}).get("global", {})
+            if global_ipv4:
+                try:
+                    requests.head("http://" + ipv4, timeout=5)
```

Comment:
I am Groot
trsnalation: ipv4 is not a string here it's a dict, so we should use global_ipv4 instead of ipv4 ?